src/riot-rs-embassy/src/arch/nrf52.rs

```diff
@@ -59,9 +59,30 @@ pub mod usb {
 
 #[cfg(feature = "hwrng")]
 pub mod hwrng {
+    use crate::arch;
+
     embassy_nrf::bind_interrupts!(pub struct Irqs {
         RNG => embassy_nrf::rng::InterruptHandler<embassy_nrf::peripherals::RNG>;
     });
+
+    pub fn construct_rng(peripherals: &mut arch::OptionalPeripherals) {
+        cfg_if::cfg_if! {
+            if #[cfg(any(context = "nrf51", context = "nrf52"))] {
+                let rng = embassy_nrf::rng::Rng::new(
+                    peripherals
+                        .RNG
+                        // We don't even have to take it out, just use it to seed the RNG
+                        .as_mut()
+                        .expect("RNG has not been previously used"),
+                    arch::hwrng::Irqs,
+                );
+
+                riot_rs_random::construct_rng(rng);
+            } else if #[cfg(context = "riot-rs")] {
```

Comment:
Why would this ever not be the case?

Reply:
when the build is not initiated by laze, e.g., just doing workspace-level `cargo  test ...`.